lib/Pipeline/Runner.cpp

```diff
@@ -190,6 +190,27 @@ Error Runner::loadFromDisk(llvm::StringRef DirPath) {
   return Error::success();
 }
 
+Error Runner::runAnalyis(llvm::StringRef AnalysisName,
+                         const ContainerToTargetsMap &Targets,
+                         llvm::raw_ostream *DiagnosticLog) {
+
+  auto MaybeStep = llvm::find_if(Steps, [AnalysisName](const auto &Step) {
+    return Step.second.getAnalysis().getName() == AnalysisName;
+  });
+
+  if (MaybeStep == Steps.end())
+    return createStringError(inconvertibleErrorCode(),
+                             "Could not find a step with a anlisys named %s\n",
+                             AnalysisName.str().c_str());
```

Comment:
* If you have multi-line statements, please use `{ }`
* Is `c_str()` necessary?

Reply:
yes, when used to do string formatting, a c string is necessary as far as i could tell.